kedro-datasets/kedro_datasets/pandas/sql_dataset.py

```diff
@@ -372,19 +377,17 @@ class SQLQueryDataset(AbstractDataset[None, pd.DataFrame]):
               date: "%Y-%m-%d %H:%M:%S.%f0 %z"
     """
 
-    # using Any because of Sphinx but it should be
-    # sqlalchemy.engine.Engine or sqlalchemy.engine.base.Engine
-    engines: Dict[str, Any] = {}
+    engines: dict[str, Engine] = {}
```

Comment:
```suggestion
    # using Any because of Sphinx but it should be
    # sqlalchemy.engine.Engine or sqlalchemy.engine.base.Engine
    engines: dict[str, Any] = {}
```